cmd/agent/subcommands/run/command.go

```diff
@@ -527,11 +546,11 @@ func stopAgent() {
 	gui.StopGUIServer()
 	profiler.Stop()
 
-	os.Remove(pidfilePath)
+	os.Remove(cliParams.pidfilePath)
```

Comment:
Do we need to guard against invocations without a specified `pidfilePath` here? My guess is not but just want to confirm.

Reply:
Good spot.  This is buggy as-is, and I haven't changed it here.  When the string is "", the call fails, but since we don't check the error return, that doesn't cause any issues.